Library/Homebrew/formulary.rb

```diff
@@ -122,9 +122,22 @@ def initialize(bottle_name)
       super name, Formulary.path(full_name)
     end
 
+    def load_file
+      formula_path = HOMEBREW_CACHE_FORMULA/"#{name}.rb"
+      formula_path.dirname.mkpath
+      FileUtils.rm_f formula_path
+      bottle_version = Utils::Bottles.resolve_version @bottle_filename
+      system "tar", "-xf", @bottle_filename, "-C", HOMEBREW_CACHE_FORMULA,
+        "--strip-components", "3", "#{name}/#{bottle_version}/.brew/#{name}.rb",
+        err: :close
```

Comment:
It would be good to avoid creating/deleting/writing files in the formulary loader. How about using something like `tar -O` to read it from `stdout`?

Reply:
Good suggestion, Mike. Done.